service/history/historyReplicator.go

```diff
@@ -1085,8 +1138,36 @@ func (r *historyReplicator) notify(clusterName string, now time.Time, transferTa
 	r.historyEngine.timerProcessor.NotifyNewTimers(clusterName, now, timerTasks)
 }
 
+func (r *historyReplicator) deserializeBlob(blob *workflow.DataBlob) ([]*workflow.HistoryEvent, error) {
+
+	if blob.GetEncodingType() != workflow.EncodingTypeThriftRW {
```

Comment:
This looks like a layering violation.  We should just use the same history serializers used by persistence layer for deserializing the payloads.  We should be able to deserialize all supported payloads like json and thrift.

Reply:
currently the new API only use ThriftRW.